app/scripts/services/tableinfo.js

```diff
@@ -104,7 +104,37 @@ angular.module('tableinfo', ['sql'])
         if (numUnassignedShards > 0 || numUnderreplicatedShards > 0) return 'warning';
         return 'good';
       }());
-
+    
+      var recovery_percent = (function() {
+        if (recovery === null || typeof recovery === 'undefined') return;
+        var init_filter = recovery.filter(function(data, idx) {
+          return data.recovery_stage === 'INIT';
+        });
+        var count_shards = init_filter.length;
+        
+        var recovered_filter = recovery.filter(function(data, idx) {
+          return data.recovery_stage !== 'INIT' && data.recovery_stage !== 'DONE';
+        });
+        count_shards += recovered_filter.length;
+        var recovered = recovered_filter.reduce(function(memo, data, idx) {
+          return memo + (data.recovery_percent / recovered_filter.length);
```

Comment:
reduce is just an iterator. this line doesn't get executed if contains 0 elements